python/pyspark/sql/session.py

```diff
@@ -305,10 +305,7 @@ def __init__(
             ):
                 jsparkSession = self._jvm.SparkSession.getDefaultSession().get()
             else:
-                jsparkSession = self._jvm.SparkSession(self._jsc.sc())
-                if options is not None:
-                    for key, value in options.items():
-                        jsparkSession.sharedState().conf().set(key, value)
+                jsparkSession = self._jvm.SparkSession(self._jsc.sc(), self._jvm.PythonUtils.toScalaMap(options))
```

Comment:
Hm, is `self._jvm.PythonUtils.toScalaMap` required?

Reply:
I think you can just do:

```python
if options is not None:
     jsparkSession = self._jvm.SparkSession(self._jsc.sc(), options)
```

Reply:
With `private[sql] def this(sc: SparkContext) = this(sc, Map.empty)` won't need such changes.

Reply:
> Hm, is `self._jvm.PythonUtils.toScalaMap` required?

It's required
```
org.apache.spark.sql.SparkSession([class org.apache.spark.SparkContext, class java.util.HashMap]) does not exist
```

Reply:
> I think you can just do:
>     if options is not None:
>           jsparkSession = self._jvm.SparkSession(self._jsc.sc(), options)

If so, we need to add a `if else`, why not just one line.